dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java

```diff
@@ -0,0 +1,200 @@
+package com.dotcms.rendering.velocity.servlet;
+
+import com.dotcms.enterprise.LicenseUtil;
+import com.dotcms.rendering.velocity.services.VelocityType;
+import com.dotcms.rendering.velocity.util.VelocityUtil;
+import com.dotcms.visitor.domain.Visitor;
+
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Identifier;
+import com.dotmarketing.beans.UserProxy;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.BlockPageCache;
+import com.dotmarketing.business.BlockPageCache.PageCacheParameters;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.factories.ClickstreamFactory;
+import com.dotmarketing.filters.CMSUrlUtil;
+import com.dotmarketing.portlets.htmlpageasset.model.IHTMLPage;
+import com.dotmarketing.portlets.rules.business.RulesEngine;
+import com.dotmarketing.portlets.rules.model.Rule;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.WebKeys;
+
+import java.io.File;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.Optional;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.apache.velocity.context.Context;
+
+import com.liferay.portal.model.User;
+
+public class VelocityLiveMode extends VelocityModeHandler {
+
+
+
+    private final HttpServletRequest request;
+    private final HttpServletResponse response;
+    private final PageMode mode = PageMode.LIVE;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this final field static too. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1170)